### ISSUE TITLE
About the training process of ur2kid

### ISSUE THREAD

**Issue by ghoshaw** [shamangary/awesome-local-global-descriptor#7]
Hi, thanks for your excellent job! 
And I have some questions about the training process of your method. 
1. first of all, your paper add global feature than D2, so your training tuple contains negative  examples, as mentioned in your paper 
![image](https://user-images.githubusercontent.com/23494873/75653231-15dfe300-5c98-11ea-92a1-1c4034bc54fa.png),my question is how much negative examples in a tuple, and did it matters a lot?
2. when you said train the network in siamese, meaning that using the tuple as input, and the loss is calculated between the example in tuple, am i right?

**Comment by shamangary** [comment]
Hello @ghoshaw, thank you for your appreciation.
1. We use 5 negative samples per tuple, but feel free to change it.
2. Siamese means that every sample in the tuple are processed by a network with the exact same weight (weight sharing). You may check https://github.com/filipradenovic/cnnimageretrieval-pytorch/blob/master/cirtorch/examples/train.py#L423 for reference.

**Comment by ghoshaw** [comment]
Thanks for your answer!
